includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2738,7 +2740,9 @@ public function update_order_status() {
 				$this->order_service->attach_transaction_fee_to_order( $order, $charge );
 			} else {
 				// For $0 orders, fetch the Setup Intent instead.
-				$intent    = $this->payments_api_client->get_setup_intent( $intent_id );
+				$setup_intent_request = Request::get( WC_Payments_API_Client::SETUP_INTENTS_API, $intent_id );
+
+				$intent    = $setup_intent_request->send( 'wcpay_get_setup_intent_request' );
```

Comment:
I think it must be like this to deal with subscription E2E test failures: 

```suggestion
				$intent    = $setup_intent_request->send( 'wcpay_get_setup_intent_request' )->to_array();
```